File: mvuab/Code/AI/StateMachine/Action.cpp
#include "Action.h"
#include "ScriptManager.h"

#include "EngineManagers.h"

CAction::CAction( CXMLTreeNode& Node )
    : m_LuaFunction( Node.GetAttribute<std::string>( "function", "no_function" ) )
{
}

CAction::~CAction()
{
}

void CAction::Execute()
{
    ScriptMInstance->RunCode( m_LuaFunction );
}

File: mvuab/bin/Videogame/Data/level2/scripts/Action.lua
class 'CActionLUA'

function CActionLUA:__init( Node )
	self.Function = Node:GetAttributeString( "function", "no_function" ).."()"
	
	engine:Trace("Loaded the action " .. self.Function)
end

function CActionLUA:Execute()
	engine:Trace("Estoy ejecutando la función ".. self.Function)
	script_manager:RunCode(self.Function)
end

function CActionLUA:GetLuaFunction()
	return self.Function
end

function CActionLUA:SetLuaFunction( LuaFunction )
	self.Function = LuaFunction
end

function CActionLUA:__tostring()
	return "Action: " .. self.Function
end